### PR TITLE
Panel: Embed URL is now correctly generated for a panel in the home dashboard

### LINKED ISSUE #37655: Embed panel iframe URL broken when sharing from home dashboard (set via config)
**What happened**:
I want to embed a panel from one of my dashboards in another site. The panel in question exists on a dashboard which is also my home dashboard, specified via config. I opened Grafana and it shows my home dashboard, I go into the panel sharing options and go to the embed tab. The iframe it gives me however does not embed that panel but the entire dashboard.

**What you expected to happen**:
To be given an iframe that embeds just the panel (and not the whole dashboard page).
Took me a while to work out but if you instead go to the dashboard directly via the manage dashboards page and then try to share the panel in the same way, it works.

Example URL generated from home (bad): http://localhost:3000/?orgId=1&refresh=5s&panelId=9

Example URL generated from dashboard's page (good): http://localhost:3000/d-solo/HbGzEKGnk/dashboard-name?orgId=1&refresh=5s&panelId=9

**How to reproduce it (as minimally and precisely as possible)**:
Interestingly this is not reproducible when setting up a test manually - i.e. creating a dashboard by hand and setting it as default. It seems to exercise this bug you need set your home dashboard using config (or in my case config via docker env vars)

I created a new dashboard, added a panel with random generated data, exported it to JSON (`test.json`). I then use that as my test dashboard for the below docker command.

```
docker run --rm -it -p3000:3000 -e GF_SECURITY_ADMIN_PASSWORD=password -v "/path/to/example/dashboards:/var/lib/grafana/dashboards" -e GF_DASHBOARDS_DEFAULT_HOME_DASHBOARD_PATH=/var/lib/grafana/dashboards/test.json grafana/grafana
```

**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.0.6 (68fe9e3431)
- Data source type & version: n/a
- OS Grafana is installed on: Official Docker container (grafana/grafana)
- User OS & Browser: Ubuntu & Chrome 92
- Grafana plugins: None
- Others: n/a

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- updates `ShareEmbed` and `ShareLink` to pass the `dashboard.uid` down
- updates `buildSoloUrl` to use the `dashboard.uid` if the route is the base route (i.e. has no dashboard uid in the existing url)
- adds a `dashboard-solo` route for snapshots/scripted dashboards
- updates + adds unit tests

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/37655

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by axelavargas** [comment]
Referencing this old issue https://github.com/grafana/grafana/issues/41736, with this change it should be fixed

**Comment by muaazsaleem** [comment]
Ah! Looking into why git is mad about my rebase.

**Comment by muaazsaleem** [comment]
Fixed the git history 😅   The CLA issue still pending.

File: public/app/features/dashboard/components/ShareModal/ShareEmbed.test.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    });
+    render(<ShareEmbed dashboard={mockDashboard} panel={mockPanel} />);
+
+    const embedUrl = screen.getByTestId('share-embed-html');
+    expect(embedUrl).toBeInTheDocument();
+    expect(embedUrl).toHaveTextContent(
+      `http://dashboards.grafana.com/dashboard-solo/script/${mockSlug}?orgId=1&from=1000&to=2000&panelId=${mockPanel.id}`
+    );
+  });
+});
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/ShareModal/ShareEmbed.test.tsx:132]: Love this!

File: public/app/routes/routes.tsx [hunk 2]
```diff
@@ -60,6 +60,14 @@ export function getAppRoutes(): RouteDescriptor[] {
...
       ),
     },
+    {
+      path: '/dashboard-solo/:type/:slug',
+      pageClass: 'dashboard-solo',
+      routeName: DashboardRoutes.Normal,
+      component: SafeDynamicImport(
+        () => import(/* webpackChunkName: "SoloPanelPage" */ '../features/dashboard/containers/SoloPanelPage')
+      ),
+    },
```
Comment (hugohaggmark) [inline public/app/routes/routes.tsx:70]: What does this new route solve? Can we add a comment?

**Review by hugohaggmark — COMMENTED** [review]
Great work 🔱  but I left some comments

File: public/app/routes/routes.tsx (hunk 2 above)
Comment (ashharrison90) [reply public/app/routes/routes.tsx:71]: have added a comment, but basically it handles the embedding of both snapshot and scripted dashboard panels

**Review by axelavargas — APPROVED** [review]
Great mob results 🔱

**Pull request merged**